code/modules/clothing/suits/hoodies.dm

```diff
@@ -1,51 +1,23 @@
 //Hoods for winter coats and chaplain hoodie etc
 
 /obj/item/clothing/suit/storage/hooded
-	var/obj/item/clothing/head/winterhood/hood
-	var/hoodtype = null
-	var/suittoggled = FALSE
 	var/opened = FALSE
 	sprite_sheets = list(
 		BODYTYPE_VAURCA_BULWARK = 'icons/mob/species/bulwark/hoodie.dmi'
 	)
+	var/hoodtype = /obj/item/clothing/head/winterhood
+	var/obj/item/clothing/head/winterhood/hood
```

Comment:
the hooded clothing shouldn't need a reference to its hood. everything the hood knows and does should be done via signals. you can use the holster style "attach proc to parent" for handling stuff like putting up the hood or taking it down, but that should also just send out signals that the hood listens to